src/protocol/session.cpp

```diff
@@ -349,7 +352,25 @@ namespace ceema {
         // Truncate MAC
         body.resize(body.size() - crypto_box_MACBYTES);
 
-        m_packetQueue.emplace_back( Packet::fromPacket(body) );
+        try
+        {
+            m_packetQueue.emplace_back( Packet::fromPacket(body) );
+        }
+        catch (const protocol_exception& e)
+        {
+            std::uintmax_t packet_type;
+            if (1 == std::sscanf(
```

Comment:
I think it might be better to just log the failed packet always, and forego the sscanf (which isnt very nice if it parses internal program data). Future unknown packets probably need to be treated the same anyway.

Reply:
Thanks for the review! I haven’t worked on any C++ projects yet that make use of exception handling, so I don’t know what makes sense here. If you think that I should simply ignore and log *all* unknown packages that’s what I’ll do.

Reply:
I implemented my suggestion in https://github.com/davidfoerster/ceema/commit/70f7961433cee19eed7303d39dd921b15a47caec to show what I mean.

Reply:
The suggestion looks great. If you can put it in this PR I'd be grateful.

There is not much in the way of convention, aside from 'throw exception when things do not work'. Back then the issue of protocol extension hadn't popped up yet, so the code does not account for it.